### ISSUE TITLE
error associating getter when using @Transient

### ISSUE THREAD

**Issue by lucasgust** [micronaut-projects/micronaut-data#2427]
### Expected Behavior

no exception

### Actual Behaviour

**CONTEXT**
removing `@Transient` annotation fixes this error but fails to persist association due to duplication of columns in the insert statement (I got this idea from [here](https://stackoverflow.com/questions/9877296/duplicate-columns-when-using-embeddedid-with-a-manytoone-mapping-with-ebean), it's a very similar problem but mine the error message is `Column 'employee_id' specified twice` and the statement is ```DEBUG DefaultR2dbcRepositoryOperations - INSERT INTO `employee_department` (`created_at`,`employee_id`,`department_id`,`employee_id`,`department_id`) VALUES (?,?,?,?,?)```)

**ACTUAL ERROR**
so I added `@Transient` annotation to be able to persist associations (and works) but now it is the read operations that is failing and giving this exception

ps.: already tried [this](https://github.com/micronaut-projects/micronaut-data/issues/942#issuecomment-1067244594)

```kotlin
WARN DefaultR2dbcRepositoryOperations - Rolling back transaction: DEFAULT on error: Constructor argument [employeeId] for type [com.example.employee.entity.EmployeeDepartmentId] must have an associated getter for dataSource default
io.micronaut.data.exceptions.MappingException: Constructor argument [employeeId] for type [com.example.employee.entity.EmployeeDepartmentId] must have an associated getter
    at io.micronaut.data.model.runtime.RuntimePersistentEntity.<init>(RuntimePersistentEntity.java:125)
    at io.micronaut.data.model.runtime.RuntimePersistentEntity.<init>(RuntimePersistentEntity.java:66)
    at io.micronaut.data.runtime.support.DefaultRuntimeEntityRegistry$1.<init>(DefaultRuntimeEntityRegistry.java:119)
    at io.micronaut.data.runtime.support.DefaultRuntimeEntityRegistry.newEntity(DefaultRuntimeEntityRegistry.java:119)
    at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1708)
    at io.micronaut.data.runtime.support.DefaultRuntimeEntityRegistry.getEntity(DefaultRuntimeEntityRegistry.java:113)
    at io.micronaut.data.runtime.support.DefaultRuntimeEntityRegistry$1.getEntity(DefaultRuntimeEntityRegistry.java:135)
    at io.micronaut.data.model.runtime.RuntimeAssociation.getAssociatedEntity(RuntimeAssociation.java:95)
    at io.micronaut.data.model.runtime.RuntimeAssociation.getAssociatedEntity(RuntimeAssociation.java:34)
    at io.micronaut.data.runtime.mapper.sql.SqlResultEntityTypeMapper$MappingContext.embeddedAssociation(SqlResultEntityTypeMapper.java:780)
    at io.micronaut.data.runtime.mapper.sql.SqlResultEntityTypeMapper$MappingContext.lambda$embedded$0(SqlResultEntityTypeMapper.java:717)
    at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1220)
    at io.micronaut.data.runtime.mapper.sql.SqlResultEntityTypeMapper$MappingContext.embedded(SqlResultEntityTypeMapper.java:717)
    at io.micronaut.data.runtime.mapper.sql.SqlResultEntityTypeMapper.readEntityId(SqlResultEntityTypeMapper.java:605)
    at io.micronaut.data.runtime.mapper.sql.SqlResultEntityTypeMapper.readEntity(SqlResultEntityTypeMapper.java:406)
    at io.micronaut.data.runtime.mapper.sql.SqlResultEntityTypeMapper.map(SqlResultEntityTypeMapper.java:197)
    at io.micronaut.data.r2dbc.operations.DefaultR2dbcRepositoryOperations$DefaultR2dbcReactiveRepositoryOperations.lambda$findAll$14(DefaultR2dbcRepositoryOperations.java:776)
    at io.micronaut.data.r2dbc.operations.DefaultR2dbcRepositoryOperations.lambda$executeAndMapEachRow$23(DefaultR2dbcRepositoryOperations.java:587)
    at dev.miku.r2dbc.mysql.MySqlResult.processRow(MySqlResult.java:176)
    at dev.miku.r2dbc.mysql.MySqlResult.handleResult(MySqlResult.java:149)
    at dev.miku.r2dbc.mysql.MySqlResult.lambda$map$1(MySqlResult.java:93)
    at reactor.core.publisher.FluxHandle$HandleSubscriber.onNext(FluxHandle.java:112)
    at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
    at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107)
    at dev.miku.r2dbc.mysql.util.DiscardOnCancelSubscriber.onNext(DiscardOnCancelSubscriber.java:70)
    at reactor.core.publisher.FluxWindowPredicate$WindowFlux.drainRegular(FluxWindowPredicate.java:670)
    at reactor.core.publisher.FluxWindowPredicate$WindowFlux.drain(FluxWindowPredicate.java:748)
    at reactor.core.publisher.FluxWindowPredicate$WindowFlux.onNext(FluxWindowPredicate.java:790)
    at reactor.core.publisher.FluxWindowPredicate$WindowPredicateMain.onNext(FluxWindowPredicate.java:268)
    at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
    at reactor.core.publisher.FluxHide$SuppressFuseableSubscriber.onNext(FluxHide.java:137)
    at reactor.core.publisher.FluxHandleFuseable$HandleFuseableSubscriber.onNext(FluxHandleFuseable.java:193)
    at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
    at reactor.core.publisher.FluxHide$SuppressFuseableSubscriber.onNext(FluxHide.java:137)
    at reactor.core.publisher.FluxContextWrite$ContextWriteSubscriber.onNext(FluxContextWrite.java:107)
    at dev.miku.r2dbc.mysql.util.DiscardOnCancelSubscriber.onNext(DiscardOnCancelSubscriber.java:70)
    at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
    at reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:200)
    at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
    at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onNext(MonoFlatMapMany.java:250)
    at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
    at reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:200)
    at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
    at reactor.core.publisher.FluxHandle$HandleSubscriber.onNext(FluxHandle.java:128)
    at io.micronaut.reactive.reactor.instrument.ReactorSubscriber.onNext(ReactorSubscriber.java:57)
    at reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:200)
    at reactor.core.publisher.EmitterProcessor.drain(EmitterProcessor.java:537)
    at reactor.core.publisher.EmitterProcessor.tryEmitNext(EmitterProcessor.java:343)
    at reactor.core.publisher.InternalManySink.emitNext(InternalManySink.java:27)
    at reactor.core.publisher.EmitterProcessor.onNext(EmitterProcessor.java:309)
    at dev.miku.r2dbc.mysql.client.ReactorNettyClient$ResponseSink.next(ReactorNettyClient.java:340)
    at dev.miku.r2dbc.mysql.client.ReactorNettyClient.lambda$new$0(ReactorNettyClient.java:103)
    at reactor.core.publisher.FluxPeek$PeekSubscriber.onNext(FluxPeek.java:185)
    at reactor.netty.channel.FluxReceive.drainReceiver(FluxReceive.java:219)
    at reactor.netty.channel.FluxReceive.onInboundNext(FluxReceive.java:353)
    at reactor.netty.channel.ChannelOperations.onInboundNext(ChannelOperations.java:353)
    at reactor.netty.channel.ChannelOperationsHandler.channelRead(ChannelOperationsHandler.java:96)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
    at dev.miku.r2dbc.mysql.client.MessageDuplexCodec.handleDecoded(MessageDuplexCodec.java:187)
    at dev.miku.r2dbc.mysql.client.MessageDuplexCodec.channelRead(MessageDuplexCodec.java:95)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:333)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:454)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
    at io.netty.handler.ssl.SslHandler.unwrap(SslHandler.java:1383)
    at io.netty.handler.ssl.SslHandler.decodeJdkCompatible(SslHandler.java:1246)
    at io.netty.handler.ssl.SslHandler.decode(SslHandler.java:1295)
    at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
    at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
    at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
    at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
    at java.base/java.lang.Thread.run(Thread.java:833)
2023-08-02T20:28:01.694Z DEBUG DefaultR2dbcRepositoryOperations - Closing Connection for DataSource: default
```

### Steps To Reproduce

1. kotlin app using micronaut data with r2dbc
2. run this migration on mysql database:
```sql
create table employee (
  id varchar(255) primary key,
  created_at datetime not null,
  name varchar(255) not null
);

create table department (
  id varchar(255) primary key,
  created_at datetime not null,
  name varchar(255) not null
);

create table employee_department (
  created_at datetime not null,
  employee_id varchar(255) primary key,
  department_id varchar(255) primary key,
  CONSTRAINT fk_employee_department_1 FOREIGN KEY (employee_id) REFERENCES employee(id),
  CONSTRAINT fk_employee_department_2 FOREIGN KEY (department_id) REFERENCES department(id)
);
```
3. populate tables
```sql
insert into employee values ('e1', now(), 'john');
insert into department values ('d1', now(), 'tech'), ('d2', now(), 'mkt');
insert into employee_department values (now(), 'e1', 'd1'), (now(), 'e1', 'd2');
```
4. mapped entities
```kotlin
import io.micronaut.data.annotation.DateCreated
import io.micronaut.data.annotation.Embeddable
import io.micronaut.data.annotation.EmbeddedId
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.MappedProperty
import io.micronaut.data.annotation.Relation
import io.micronaut.data.annotation.Relation.Kind.MANY_TO_ONE
import io.micronaut.data.annotation.Transient
import io.micronaut.data.jdbc.annotation.JoinColumn
import java.time.LocalDateTime

@MappedEntity("employee")
data class EmployeeEntity(

    @field:Id
    val id: String,

    @DateCreated
    var createdAt: LocalDateTime? = null,

    var name: String
)

@MappedEntity("department")
data class DepartmentEntity(

    @field:Id
    val id: String,

    @DateCreated
    var createdAt: LocalDateTime? = null,

    var name: String
)

@Embeddable
data class EmployeeDepartmentId(

    @MappedProperty("employee_id")
    @Transient
    val employeeId: String,

    @MappedProperty("department_id")
    @Transient
    val departmentId: String
)

@MappedEntity("employee_department")
data class EmployeeDepartmentEntity(

    @DateCreated
    var createdAt: LocalDateTime? = null,

    @EmbeddedId
    val id: EmployeeDepartmentId,

    @Relation(value = MANY_TO_ONE)
    @JoinColumn(name = "employee_id")
    val employee: EmployeeEntity,

    @Relation(value = MANY_TO_ONE)
    @JoinColumn(name = "department_id")
    val department: DepartmentEntity
)
```
5. define repositories
```kotlin
import io.micronaut.data.annotation.Join
import io.micronaut.data.annotation.Join.Type.LEFT_FETCH
import io.micronaut.data.annotation.repeatable.JoinSpecifications
import io.micronaut.data.model.query.builder.sql.Dialect.MYSQL
import io.micronaut.data.r2dbc.annotation.R2dbcRepository
import io.micronaut.data.repository.jpa.kotlin.CoroutineJpaSpecificationExecutor
import io.micronaut.data.repository.kotlin.CoroutineCrudRepository

@R2dbcRepository(dialect = MYSQL)
interface EmployeeRepository :
    CoroutineCrudRepository<EmployeeEntity, String>,
    CoroutineJpaSpecificationExecutor<EmployeeEntity>

@R2dbcRepository(dialect = MYSQL)
interface DepartmentRepository :
    CoroutineCrudRepository<DepartmentEntity, String>,
    CoroutineJpaSpecificationExecutor<DepartmentEntity>

@R2dbcRepository(dialect = MYSQL)
interface EmployeeDepartmentRepository :
    CoroutineCrudRepository<EmployeeDepartmentEntity, EmployeeDepartmentId>,
    CoroutineJpaSpecificationExecutor<EmployeeDepartmentEntity> {
    @JoinSpecifications(
        Join(value = "employee"),
        Join(value = "department", type = LEFT_FETCH)
    )
    override fun findAll(): Flow<EmployeeDepartmentEntity>
}
```
6. find all employee department associations
```kotlin
class SomeAdapterClass(
    private val repository: EmployeeDepartmentRepository
) {
    suspend fun findAllDepartments(employeeId: String): List<Department> {
        val employeeDepartment = repository.findAll() // --> error

        // ...
    }
}
```

### Environment Information

macOS Ventura 13.4.1
JDK 17.0.7

### Version

3.9.4

**Comment by radovanradic** [comment]
Try mapping like this `EmployeeDepartmentId`
```
@Embeddable
data class EmployeeDepartmentId(

    @Relation(Relation.Kind.MANY_TO_ONE)
    @MappedProperty("employee_id")
    val employee: Employee,

    @Relation(Relation.Kind.MANY_TO_ONE)
    @MappedProperty("department_id")
    val department: Department
)
```
and `EmployeeDepartment`
```
@MappedEntity
data class EmployeeDepartment(

    @EmbeddedId
    val id: EmployeeDepartmentId,

    @DateCreated
    val createdAt: LocalDateTime? = null
)
```
Maybe I didn't get Kotlin classes right, but here is small demo app in Java and H2 db that shows how it could be done
[mn-embeddedid-demo.zip](https://github.com/micronaut-projects/micronaut-data/files/12248022/mn-embeddedid-demo.zip)

**Comment by lucasgust** [comment]
@radovanradic it works but then I start to have a problem retriving data if association also has a many-to-many association (this is my case!) and this forced me to return to the way it is described in my initial question

let's update this example supposing an employee also has many roles and we want to create an API that get a department by id with all employees and its roles associated

increment this in steps 2 and 3
```sql
create table role (
  id varchar(255) primary key,
  created_at datetime not null,
  name varchar(255) not null
);

create table employee_role (
  created_at datetime not null,
  employee_id varchar(255) primary key,
  role_id varchar(255) primary key,
  CONSTRAINT fk_employee_role_1 FOREIGN KEY (employee_id) REFERENCES employee(id),
  CONSTRAINT fk_employee_role_2 FOREIGN KEY (role_id) REFERENCES role(id)
);

insert into role values ('r1', now(), 'manager'), ('r2', now(), 'analist');
insert into employee_role values (now(), 'e1', 'r1'), (now(), 'e1', 'r2');
```

update step 4
```kotlin
import io.micronaut.data.annotation.DateCreated
import io.micronaut.data.annotation.Embeddable
import io.micronaut.data.annotation.EmbeddedId
import io.micronaut.data.annotation.Id
import io.micronaut.data.annotation.MappedEntity
import io.micronaut.data.annotation.MappedProperty
import io.micronaut.data.annotation.Relation
import io.micronaut.data.annotation.Relation.Kind.MANY_TO_ONE
import io.micronaut.data.annotation.Transient
import io.micronaut.data.jdbc.annotation.JoinColumn
import java.time.LocalDateTime

@MappedEntity("employee")
data class EmployeeEntity(

    @field:Id
    val id: String,

    @DateCreated
    var createdAt: LocalDateTime? = null,

    var name: String,

    @Relation(value = ONE_TO_MANY, mappedBy = "employee")
    val roleAssociation: List<EmployeeRoleEntity> = emptyList()
)

@MappedEntity("department")
data class DepartmentEntity(

    @field:Id
    val id: String,

    @DateCreated
    var createdAt: LocalDateTime? = null,

    var name: String,

    @Relation(value = ONE_TO_MANY, mappedBy = "department")
    val employeeAssociation: List<EmployeeDepartmentEntity> = emptyList()
)

@MappedEntity("role")
data class RoleEntity(

    @field:Id
    val id: String,

    @DateCreated
    var createdAt: LocalDateTime? = null,

    var name: String
)

@Embeddable
data class EmployeeDepartmentId(

    @Relation(value = MANY_TO_ONE)
    @MappedProperty("employee_id")
    val employee: EmployeeEntity,

    @Relation(value = MANY_TO_ONE)
    @MappedProperty("department_id")
    val department: DepartmentEntity
)

@MappedEntity("employee_department")
data class EmployeeDepartmentEntity(

    @DateCreated
    var createdAt: LocalDateTime? = null,

    @EmbeddedId
    val id: EmployeeDepartmentId
)

@Embeddable
data class EmployeeRoleId(

    @Relation(value = MANY_TO_ONE)
    @MappedProperty("employee_id")
    val employee: EmployeeEntity,

    @Relation(value = MANY_TO_ONE)
    @MappedProperty("role_id")
    val role: RoleEntity
)

@MappedEntity("employee_role")
data class EmployeeRoleEntity(

    @DateCreated
    var createdAt: LocalDateTime? = null,

    @EmbeddedId
    val id: EmployeeRoleId
)

```

update step 5

```kotlin
import io.micronaut.data.annotation.Join
import io.micronaut.data.annotation.Join.Type.LEFT_FETCH
import io.micronaut.data.annotation.repeatable.JoinSpecifications
import io.micronaut.data.model.query.builder.sql.Dialect.MYSQL
import io.micronaut.data.r2dbc.annotation.R2dbcRepository
import io.micronaut.data.repository.jpa.kotlin.CoroutineJpaSpecificationExecutor
import io.micronaut.data.repository.kotlin.CoroutineCrudRepository

@R2dbcRepository(dialect = MYSQL)
interface DepartmentRepository :
    CoroutineCrudRepository<DepartmentEntity, String>,
    CoroutineJpaSpecificationExecutor<DepartmentEntity> {

    @Join(value = "employeeAssociation.id.employee.roleAssociation.id.role", type = LEFT_FETCH)
    override suspend fun findById(id: UUID): DepartmentEntity?
}
```

finally, when I call `departmentRepository.findById(id)` employee roles return empty. when I executed logged query directly from database it gives all data so I guess there is an error mapping columns from ResultSet

actual response:
```json
{
    "department": {
        "id": "d1",
        "name": "tech",
        "createdAt": "2000-01-01",
        "employees": [
            "id": "e1",
            "name": "john",
            "createdAt": "2000-01-01",
            "roles": []
        ]
    }
}
```

expected response:
```json
{
    "department": {
        "id": "d1",
        "name": "tech",
        "createdAt": "2000-01-01",
        "employees": [
            "id": "e1",
            "name": "john",
            "createdAt": "2000-01-01",
            "roles": [
                {
                    "id": "r1",
                    "name": "manager",
                    "createdAt": "2000-01-01",
                },
                {
                    "id": "r2",
                    "name": "analist",
                    "createdAt": "2000-01-01",
                }
            ]
        ]
    }
}
```

**Comment by radovanradic** [comment]
Ok, I thought the issue is with `@Transient` and focused on that.
Here is another example of mapping that could resolve your issue, unless there is some other scenario that might not be suitable with this.
[mn-embeddedid-demo.zip](https://github.com/micronaut-projects/micronaut-data/files/12259486/mn-embeddedid-demo.zip)

**Comment by lucasgust** [comment]
@radovanradic sorry, the main problem was the use of `@Transient`. I only wanted to give you context on why I didn't leave many to one associations properties inside Embedded class (as mentioned, due to join problems).

following your code I managed to make it work because I found out the reason why it wasn't working here. the suffix of my classes is `Entity` (differing from the table names) and this seems to "confuse" query with a `@Join` to a association table (by putting an extra `_entity` after entity names, for example `employee_entity_department_entity`). I was only able to get it to work when I changed the class names to match the table names (would this be a bug?) like your code.

however, the use of `@Transient` appears to be a bug. fortunately I no longer depend on it to continue with my development.

**Comment by radovanradic** [comment]
Not sure what you expect from `@Transient`, but when you mark the field with it then the field becomes unknown for the persistence layer - it won't be used for writes or reads from the database. So, I think that's not the issue but mapping configuration that was not correct.
Also, if mapping is 
```
@MappingEntity
public class EmployeeEntity {
```
then Micronaut by default expects table will be named `employee_entity`. You can configure it like
```
@MappingEntity("employee")
public class EmployeeEntity {
```
so Micronaut will expect table to be named `employee`.

I think we can close this ticket and if you face another issue, please reopen new one with the details and potentially reproducer.

Thanks

**Comment by lucasgust** [comment]
@radovanradic 

> Not sure what you expect from @Transient

as I said in the initial question and posted the stackTrace, I don't expect an error: `Constructor argument [employeeId] for type [com.example.employee.entity.EmployeeDepartmentId] must have an associated getter` (I also encountered a similar [open issue](https://github.com/micronaut-projects/micronaut-data/issues/942#issuecomment-1067244594) , but that solution didn't work for me)

>you can configure it like `@MappingEntity("employee")` so Micronaut will expect table to be named employee

as I also said, the problem is when you have an association table involved. using your [own example](https://github.com/micronaut-projects/micronaut-data/files/12259486/mn-embeddedid-demo.zip)), I was able to reproduce the error by following the steps I gave. if you interessed, rename these classes `Employee -> EmployeeEntity`, `Department -> DepartmentEntity` and `DepartmentEmployee -> DepartmentEmployeeEntity` (and put table name in the `@MappingEntity` value on each) and run your test. the error is the same I mentioned (`Table 'test.department_entity_employee_entity' doesn't exist`) so I guess there is a bug
